components/style/properties/gecko.mako.rs

```diff
@@ -1390,33 +1245,15 @@ fn static_assert() {
                                                                 .mLayers.iter_mut()) {
             % if shorthand == "background":
                 if let Some(image) = image.0 {
-                    match image {
-                        Image::Gradient(gradient) => {
-                            set_gradient(gradient, &mut geckoimage.mImage)
-                        },
-                        Image::Url(..) => {
-                            // let utf8_bytes = url.as_bytes();
-                            // Gecko_SetUrlImageValue(&mut self.gecko.mImage.mLayers.mFirstElement,
-                            //                        utf8_bytes.as_ptr() as *const _,
-                            //                        utf8_bytes.len());
-                            warn!("stylo: imgRequestProxies are not threadsafe in gecko, \
-                                   background-image: url() not yet implemented");
-                        }
-                    }
+                    set_image(image, &mut geckoimage.mImage)
```

Comment:
@Manishearth  It looks like else branch has only one more match pattern than if branch. And I kept that match in else part. Moved the inner match pattern which looks pretty same. In if branch, `Image::Url` prints a warning about url and in else branch, it does nothing except gradient(with this line `_ => () // we need to support image values`) I thought we can give a warning in else branch too since url's not implemented in both. Should I check this still?